src/modules/hide_vodcasts/index.js

```diff
@@ -0,0 +1,31 @@
+const $ = require('jquery');
+const settings = require('../../settings');
+const watcher = require('../../watcher');
+
+const VODCAST_SELECTOR = '.live-channel-card .tw-pill';
+const CARD_SELECTOR = '.live-channel-card';
+
+class HideVodcastsModule {
+    constructor() {
+        settings.add({
+            id: 'hideVodcasts',
+            name: 'Hide Vodcasts',
+            defaultValue: false,
+            description: 'Hide Vodcasts from directory listings'
+        });
+        settings.on('changed.hideVodcasts', value => value === true ? this.hide() : this.show());
+
+        watcher.on('directory.vodcast', () => this.hide());
+    }
+
+    hide() {
+        if (settings.get('hideVodcasts') === false) return;
+        $(VODCAST_SELECTOR).closest(CARD_SELECTOR).parent(':visible').hide();
```

Comment:
it would be nice if we can avoid a mutation observer.. i dont see vodcast pills so either they disabled them or it isnt fully rolled out yet.. without seeing them myself i cant tell you if this is the best way

Reply:
The mutation observer was used since there's a loading icon before live channels are displayed and it's what I had used in the pre new-twitch code.